lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -128,7 +128,14 @@ <h3 class="course-title" id="course-title-${enrollment.course_id}">
           %>
 
             <span class="info-date-block-container">
-                % if is_unfulfilled_entitlement:
+                % if not show_courseware_link and not is_unfulfilled_entitlement and hasattr(show_courseware_link, 'user_message') and hasattr(show_courseware_link, 'error_code') and show_courseware_link.error_code == 'audit_expired':
```

Comment:
@cpennington it looks like there is some code that you already committed that just checks for the presence of a user_message and not the error_code. It looks like there are other errors that have a user_message which were appearing without this condition, so not sure if that will be a problem

Reply:
Stylistically, you could use `getattr(show_courseware_link, 'error_code') == 'audit_expired'`.

However, my intention was that this would actually display the user_message for *any* access denied error that floated up to the dashboard, rather than having it be specific to this particular code. If we need to change the UX a bit to support that added level of generality, I'd be in favor of that.

Reply:
that makes sense from a technical perspective, but given that there are preexisting user_messages that would appear that product is probably not aware of, this seems like something product and UX should decide

Reply:
@sstack22  @cpennington 
Some of the current access response error messages are below.
I think the question is, do we want to remove the View Course button and replace the start date with the message if you don't have access to the course due to one of the errors below.

Course has not started
Course does not start until ...
You have unfulfilled milestones
You do not have access to this course
You do not have access to this course on a mobile device

Reply:
Have you looked at which other `AccessResponse`s have error messages that could possibly end up on the dashboard?

Reply:
I know at least the start date ones could show up there because I encountered that in a local bokchoy test.  
But theoretically I would imagine all the ones I listed can (except for the mobile app error since the code to expose the errors is not touching that)

Reply:
@MatthewPiatetsky - hmm, do we need to define error the behavior for other errors at this point? I'm not sure clear on the question. The 'View Course' button should not be available if your access has expired. If your access has not expired, and one of those error occur, I'd expect what happens today to continue happening.

Reply:
@sstack22 the mocks put the expiration message where the start date goes. the question is whether we want to replace the start date with the message (and hide the view course button) if the user can't access the course for one of the other reasons in my list.  
If we want what currently happens to happen, then we wouldn't want any of those errors to change anything, but I want to make sure you understand the question

Reply:
@MatthewPiatetsky - I may still be confused, but let me know if this answers your question. The dashboard date message already has different states:

1. Starts (in the future) - View Course Button is not available
2. Started
3. Ended - this ended message replaces the started message

I would not add additional messaging for the other errors (like milestones). The only change we want is to show "Expired" when access expired, in place of the start/end date messaging - so, adding a 4th state.

Reply:
ok. @cpennington so it sounds to me like we don't want to display any other errors, so I will keep the audit_expired check. I can change it to the different code you suggested though.

Reply:
@sstack22 fyi this is what happens
<img width="946" alt="screen shot 2018-10-17 at 2 39 07 pm" src="https://user-images.githubusercontent.com/5958221/47108804-92486200-d21a-11e8-91cd-89ff1c78558c.png">

I think @cpennington was of the opinion that a consistent way to show why you don't have access to a course would be good.
I personally don't care, but I do need to know what to implement
I've made it so that message will not appear in the current state of the pull request, but I can change it if necessary